### PR TITLE
Revert AliasTuple back to AliasSeq

### PR DESCRIPTION
**Pull request by JinShil**
See http://forum.dlang.org/post/mo8jrn$1kro$1@digitalmars.com

### CODE REVIEW

**Comment by quickfur** [comment]
LGTM. Seems @andralex approves in the forum thread, so I'm tempted to toggle automerge, but I'll let a few more reviewers chime in first.

**Comment by schveiguy** [comment]
I'm really done with this debate, but I'll at least chime in and say I still have no objection to `AliasSeq`.

**Comment by DmitryOlshansky** [comment]
The leadearship has spoken. I wouldn't merge simpy because that's be me merging the same thing twice.

**Comment by schuetzm** [comment]
-1 from, for reasons already explained several times. It's the worst of all suggestions so far.

**Comment by MartinNowak** [comment]
People argued against pack b/c AliasSeq auto-expands.
In that sense it's very different from http://en.cppreference.com/w/cpp/language/parameter_pack.

File: std/meta.d [hunk 1]
```diff
@@ -31,19 +31,19 @@
 module std.meta;
 
 /**
- * Creates a list of zero or more aliases. This is most commonly
+ * Creates a a sequence of zero or more aliases. This is most commonly
```
Comment (MartinNowak) [inline std/meta.d]: Typo

File: std/meta.d [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 /**
- * Templates to manipulate template argument lists (also known as alias tuples).
+ * Templates to manipulate template argument lists (also known as type lists).
  *
- * Some operations on alias tuples are built in to the language,
- * such as TL[$(I n)] which gets the $(I n)th alias from the
- * alias tuple. TL[$(I lwr) .. $(I upr)] returns a new alias
- * tuple that is a slice of the old one.
+ * Some operations on alias sequences are built in to the language,
+ * such as TL[$(I n)] which gets the $(I n)th type from the
```
Comment (MartinNowak) [inline std/meta.d]: Please don't use type here. Either stick with alias or use something generic such as entry or element.

File: std/meta.d [hunk 3]
```diff
@@ -1,12 +1,12 @@
 // Written in the D programming language.
 
 /**
- * Templates to manipulate template argument lists (also known as alias tuples).
+ * Templates to manipulate template argument lists (also known as type lists).
```
Comment (MartinNowak) [inline std/meta.d]: Next paragraph should state that this can contain types, symbols (aliases), and values.

File: std/meta.d [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
  *
- * Some operations on alias tuples are built in to the language,
- * such as TL[$(I n)] which gets the $(I n)th alias from the
- * alias tuple. TL[$(I lwr) .. $(I upr)] returns a new alias
- * tuple that is a slice of the old one.
+ * Some operations on alias sequences are built in to the language,
+ * such as TL[$(I n)] which gets the $(I n)th type from the
+ * alias sequence. TL[$(I lwr) .. $(I upr)] returns a new type
+ * list that is a slice of the old one.
```
Comment (MartinNowak) [inline std/meta.d]: Why not AliasSeq?

File: std/meta.d [hunk 5]
```diff
@@ -54,26 +54,26 @@ unittest
 ///
 unittest
 {
-    alias TL = AliasTuple!(int, double);
+    alias TL = AliasSeq!(int, double);
```
Comment (MartinNowak) [inline std/meta.d]: Replace TL with AS?

**Comment by MartinNowak** [comment]
Why do you use TypeList/TList throughout the documentation?

File: std/meta.d [hunk 6]
```diff
@@ -1018,7 +1018,7 @@ unittest
  * Instantiates the given template with the given list of parameters.
  *
  * Used to work around syntactic limitations of D with regard to instantiating
- * a template from an alias tuple (e.g. T[0]!(...) is not valid) or a template
+ * a template from a alias sequence (e.g. T[0]!(...) is not valid) or a template
```
Comment (MartinNowak) [inline std/meta.d]: Typo s/a/an/.

File: std/meta.d [hunk 7]
```diff
@@ -648,28 +648,28 @@ unittest
 
 
 /**
- * Filters a $(D AliasTuple) using a template predicate. Returns a
- * $(D AliasTuple) of the elements which satisfy the predicate.
+ * Filters a $(D AliasSeq) using a template predicate. Returns a
```
Comment (MartinNowak) [inline std/meta.d]: an

File: std/meta.d [hunk 8]
```diff
@@ -648,28 +648,28 @@ unittest
...
+ * $(D AliasSeq) of the elements which satisfy the predicate.
  */
-template Filter(alias pred, TTuple...)
+template Filter(alias pred, TList...)
```
Comment (MartinNowak) [inline std/meta.d]: TList again, should work with any AliasSeq.

**Comment by andralex** [comment]
I'm fine with this. If no strong new objection please merge but only after minding @MartinNowak's review.

**Comment by JinShil** [comment]
@MartinNowak Thanks for the thorough review.  I fixed the typos, but I'm not the person to fix the irregularities in the naming conventions here as I think there are quite a few more than the ones you pointed out.  

For the record, I'm simply reverting https://github.com/D-Programming-Language/phobos/pull/3488 (which was merged prematurely) back to https://github.com/D-Programming-Language/phobos/pull/3463.

I politely ask that you merge this PR as is, putting master back into a previously merged state, and that @deadalnix (originator of https://github.com/D-Programming-Language/phobos/pull/3463) please volunteer to clean up the naming conventions.

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Comment by andralex** [comment]
thx all!

**Comment by WalterBright** [comment]
It's ironic that `AliasSeq` was selected, whereas `setExt` was vilified for using an abbreviation.

**Comment by DmitryOlshansky** [comment]
@WalterBright keeping in mind that there already was setExtension adding setExt met expected resistance.

**Comment by WalterBright** [comment]
@klickverbot so what do you think of Seq?

**Comment by JinShil** [comment]
@WalterBright very few are happy with this name.  It was selected by decree and out of frustration because the conversation devolved into sarcasm and belittlement of cooperation as "bikeshedding"

**Pull request merged**